examples/auth.rs

```diff
@@ -0,0 +1,81 @@
+#![allow(clippy::not_unsafe_ptr_arg_deref)]
+
+use mosquitto_plugin::*;
+use std::collections::{HashMap, HashSet};
+
+#[derive(Debug, Default)]
+#[allow(dead_code)]
+pub struct AuthPlugin {
+    authenticated: HashSet<String>,
+}
+
+// Required trait implementation
+impl MosquittoPlugin for AuthPlugin {
+    fn init(_opts: HashMap<&str, &str>) -> Self {
+        AuthPlugin::default()
+    }
+
+    fn on_disconnect(&mut self, client: &dyn MosquittoClientContext, _reason: i32) {
+        println!(
+            "Plugin on_disconnect, Client {:?} is disconnecting",
+            client.get_id()
+        );
+    }
+
+    fn acl_check(
+        &mut self,
+        client: &dyn MosquittoClientContext,
+        _acl: AclCheckAccessLevel,
+        _msg: MosquittoMessage,
+    ) -> Result<Success, Error> {
+        println!("Plugin on_acl_check, Client {:?}", client.get_id());
+        let client_id = client.get_id().ok_or(Error::Inval)?;
+
+        if self.authenticated.contains(&client_id) {
```

Comment:
Is this really necessary? Would not authentication not allow the client to connect even? therefore prohibiting any sub/pub or similar?

This might spark wrongful useage if that is the case

Reply:
My intention is that a client that a client without client_id cannot be authenticated at all.  I'll update the example

Reply:
I did not know that clients could connect without a client_id. 

If this is because the client_id is optional in the trait mosquittoclientcontext, that is because before auth, the client is not guaranteed to have a client_id, but after auth, it must be there. At least from what I have gathered. 

Basically, I believe that the code you now have, is equal to always ` Ok(Success)`, but with extra capacity for bugs (since you can forget to remove from the list etc